### PR TITLE
Ensure all SCP & HCA query results appear in bulk download modal (SCP-3911)

### PR DESCRIPTION
**Pull request by bistline**
Currently, when running a cross-dataset search query that returns more than 1 page of results, there is a bug where only the HCA results shown on the current page will appear in the bulk download modal, regardless of how many total entries were found.  This update addresses that issue by including both SCP & HCA accessions in the `BulkDownloadController#summary` endpoint, and running an additional query on Azul to retrieve summary file info for HCA data.

MANUAL TESTING
1. Boot as normal, sign in with an account that has `cross_dataset_search_backend` enabled
2. Run a faceted search for `species:Homo sapiens, Mus musculus` and `organ:lung, stomach`
3. Confirm that there are more than 1 page of results (there should be at least 16 HCA entries)
4. Click `Download`, and confirm that all HCA entries appear in the table (the screenshot below is not the entire list, but you can see that there are more than 10):
Screen Shot 2021-12-10 at 10 29 12 AM

This PR satisfies SCP-3911.

### CODE REVIEW

File: app/javascript/components/search/controls/download/DownloadSelectionModal.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React, { useEffect, useState } from 'react'
 import Modal from 'react-bootstrap/lib/Modal'
 import _cloneDeep from 'lodash/cloneDeep'
+import { partition } from 'lodash'
```
Comment (devonbush) [inline app/javascript/components/search/controls/download/DownloadSelectionModal.js:4]:
```suggestion
import _partition from 'lodash/partition'
```
The underscore prefix makes it clearer that this is a lodash method, and also utilizes lodash's partial import capability (so the import only grabs exactly the lodash fragment you need)

File: app/javascript/components/search/controls/download/DownloadSelectionModal.js [hunk 2]
```diff
@@ -41,8 +41,12 @@ export default function DownloadSelectionModal({ studyAccessions, azulFileInfo,
     render bulk download table for SCP & HCA studies
    */
   function renderFileTables(result=[]) {
-    setSelectedBoxes(newSelectedBoxesState(result, SCP_COLUMNS))
-    setDownloadInfo(result)
+    const partitionedResults = partition(result, ['studySource', 'SCP'])
+    const scpFileInfo = partitionedResults[0]
+    const azulFileInfo = partitionedResults[1]
```
Comment (devonbush) [inline app/javascript/components/search/controls/download/DownloadSelectionModal.js:46]:
```suggestion
    const [scpFileInfo, azulFileInfo]  = _partition(result, ['studySource', 'SCP'])
```
using a destructured assignment for clarity/conciseness

File: app/lib/azul_search_service.rb [hunk 3]
```diff
@@ -201,4 +201,37 @@ def self.extract_file_information(result)
...
   end
+
+  # query Azul with project shortnames and return summary file information
+  # this method mirrors interface on BulkDownloadService#study_download_info for use in bulk download modal
+  def self.get_file_summary_info(accessions)
+    client = ApplicationController.hca_azul_client
+    file_summaries = []
+    file_query = { 'project' => { 'is' => accessions } }
+    results = client.projects(query: file_query)
+    results['hits'].each do |entry|
```
Comment (devonbush) [inline app/lib/azul_search_service.rb:212]:
```suggestion
    file_summaries = results['hits'].map do |entry|
```

File: app/lib/azul_search_service.rb [hunk 4]
```diff
@@ -201,4 +201,37 @@ def self.extract_file_information(result)
...
+            file_type: 'Project Manifest',
+            count: 1,
+            upload_file_size: 1.megabyte, # placeholder filesize as we don't know until manifest is downloaded
+            name: "#{accession}.tsv"
+          }
+        ]
+      }.with_indifferent_access
+      project_file_info = extract_file_information(entry_hash)
+      result[:studyFiles] += project_file_info if project_file_info.any?
+      file_summaries << result
```
Comment (devonbush) [inline app/lib/azul_search_service.rb:233]:
```suggestion
     result
```
if using `map` as suggested above

**Review by devonbush — APPROVED** [review]
I really like how you were able to accomplish this with minimal architecture changes.  A few codesmithing suggestions.  The only blocker is the `lodash` import -- using the scoped import lets lodash optimize our JS bundle better.

**Review by devonbush — CHANGES_REQUESTED** [review]
oops, clicked approve when I meant to request changes :)

File: app/javascript/components/search/controls/download/DownloadSelectionModal.js (hunk 2 above)
Comment (bistline) [reply app/javascript/components/search/controls/download/DownloadSelectionModal.js:46]: aah I wasn't sure if that worked in JS - nice.

File: app/lib/azul_search_service.rb (hunk 4 above)
Comment (bistline) [reply app/lib/azul_search_service.rb]: 39fe8e0

File: app/lib/azul_search_service.rb (hunk 3 above)
Comment (bistline) [reply app/lib/azul_search_service.rb]: 39fe8e0

File: app/javascript/components/search/controls/download/DownloadSelectionModal.js (hunk 2 above)
Comment (bistline) [reply app/javascript/components/search/controls/download/DownloadSelectionModal.js]: 39fe8e0

File: app/javascript/components/search/controls/download/DownloadSelectionModal.js (hunk 1 above)
Comment (bistline) [reply app/javascript/components/search/controls/download/DownloadSelectionModal.js]: 39fe8e0

**Review by devonbush — APPROVED** [review]

File: app/javascript/components/search/controls/download/DownloadSelectionModal.js [hunk 5]
```diff
@@ -41,8 +40,10 @@ export default function DownloadSelectionModal({ studyAccessions, azulFileInfo,
     render bulk download table for SCP & HCA studies
    */
   function renderFileTables(result=[]) {
-    setSelectedBoxes(newSelectedBoxesState(result, SCP_COLUMNS))
-    setDownloadInfo(result)
+    const [scpFileInfo, azulFileInfo] = _partition(result, ['studySource', 'SCP'])
```
Comment (eweitz) [inline app/javascript/components/search/controls/download/DownloadSelectionModal.js:43]: [`partition`](https://lodash.com/docs/4.17.15#partition) looks neat!  Today I learned.

**Review by eweitz — APPROVED** [review]
Code looks good!

**Pull request merged**